### ISSUE TITLE
Tracking the first timestamp in h-test

### ISSUE THREAD

**Issue by stompsjo** [CNERG/RadClass#21]
I propose the following introduction of `self.x1_timestamp` into the run methods of `H0` in #18 (or its eventually conclusion should the PR be merged first):
```
def run_gross(self, data, timestamp):
        data = np.sum(data)

        # only needed for the first initialization
        if self.x1 is None:
            self.x1_timestamp = timestamp
            self.x1 = data
        else:
            self.x2 = data
            n = self.x1 + self.x2
            p = 0.5
            pval = stats.binom_test(self.x1, n, p, alternative='two-sided')

            # only save instances with rejected null hypothesesf
            if pval <= self.significance:
                self.triggers = np.append(self.triggers,
                                            [[self.x1_timestamp, pval,
                                            self.x1, self.x2]],
                                            axis=0)

            # saving data for the next integration step
            self.x1 = self.x2
            self.x1_timestamp = timestamp
```
A subsequent change would also be made to `run_channel`. This allows `H0` to track the timestamp of `x1` rather than `x2` if the null hypothesis is rejected. It has been some indication in testing that the first timestamp may be more important to keep track of, in the event that `x1 > x2`.

**Further**, it may be better to create some `if...then` test to save the timestamp of the observation (`x1` or `x2`) that is larger. That way, the timestamp used to find the spectrum of the corresponding higher count rate is saved. I think this is a design question that could benefit from some discussion.

**Comment by gonuke** [comment]
This might be better as a PR. Including blocks of code in an issue doesn't really allow us to see a comparison of the code.

**Comment by gonuke** [comment]
The issue is a better place to make a note of what feature is being requested or bug being reported, with a subsequent PR showing the necessary/proposed code changes.

**Comment by stompsjo** [comment]
Agreed. I probably should have just made a branch on my fork without PR'ing it until PR #18 was merged. I just didn't want confusion about what needed rebasing, etc. but wanted a public place where I wouldn't forget my suggestion 😄  Starting a PR now!

**Comment by stompsjo** [comment]
Now implemented in #22
